src/fits_schema/binary_table.py

```diff
@@ -165,51 +165,78 @@ def validate_data(self, data, onerror="raise"):
         try:
             # casting = 'safe' makes sure we don't change values
             # e.g. casting doubles to integers will no longer work
-            data = np.asanyarray(data).astype(self.dtype, casting="safe")
+
+            if np.dtype(data.dtype).char in ["S", "U"]:
+                # special case is if this is a string column, i.e. a Char field
+                # with shape=None. There we should ignore the length of the
+                # numpy dtype (e.g. 'S12' should be checked as 'S'). Note that
+                # inside a FITS file, the strings are stored as dtype('S')
+                # always, however if you call hdu.data[column], they are
+                # magically converted to 'U' by astropy, but not for astropy
+                # Tables or np.recarrays (why is not obvious).
+                data = np.asanyarray(data).astype("U", casting="safe")
+
+                # string data must be ascii only:
+                if not all(s.isascii() for s in data.ravel()):
```

Comment:
```

In [10]: np.array(["Nöthe"]).astype("S")
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
Cell In[10], line 1
----> 1 np.array(["Nöthe"]).astype("S")

UnicodeEncodeError: 'ascii' codec can't encode character '\xf6' in position 1: ordinal not in range(128)
```
should be much faster

Reply:
https://numpy.org/devdocs/user/basics.strings.html#casting-to-and-from-fixed-width-strings